pkg/client/file.go

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package client
+
+//go:generate ../../bin/mockgen -copyright_file=../../hack/copyright_header.txt -destination=./mocks/mock_file.go -package=mocks github.com/Azure/aks-tls-bootstrap-client/pkg/client FileReader
+
+import "os"
+
+// fileReader proves methods to interact with the host file system
+// needed by the bootstrap server in order to properly perform setup.
+// This is primarily used for mockability within unit tests.
+type FileReader interface {
```

Comment:
we shouldn't need to have this be an exported interface, you can use mockgen to mock unexported interfaces with exported methods: https://github.com/golang/mock/issues/297#issuecomment-577940136